internal/cmd/shell.go

```diff
@@ -178,15 +175,36 @@ func runCommand(ctx context.Context, envName string, command string, args []stri
 	}()
 	err = process.Wait()
 	if err != nil {
-		var closeErr websocket.CloseError
-		if xerrors.Is(err, ctx.Err()) || xerrors.As(err, &closeErr) {
-			return xerrors.Errorf("network error, is %q online?", envName)
-		}
-		return err
+		return prettyShellExitError(ctx, err, entClient, env.Name)
 	}
 	return nil
 }
 
+func prettyShellExitError(ctx context.Context, err error, client *coder.Client, envName string) error {
+	var closeErr websocket.CloseError
+	if xerrors.Is(err, ctx.Err()) || xerrors.As(err, &closeErr) {
+		if err != nil {
```

Comment:
we already check that the error is not nil before calling the function., no need to check again.
Would be better to add a comment on the function stating that the error passed to it is expected to be set, makes things easier to follow and avoid superfluous checks by simplifying the code.